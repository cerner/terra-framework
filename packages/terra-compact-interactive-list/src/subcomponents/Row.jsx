import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import ThemeContext from 'terra-theme-context';
import classNames from 'classnames/bind';
import styles from './Row.module.scss';
import Cell from './Cell';
import cellShape from '../proptypes/cellShape';
import formattedColumnShape from '../proptypes/formattedColumnShape';
import { widthUnitTypes } from '../utils/constants';
import unitShape from '../proptypes/unitShape';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * An identifier to uniquely identify the row.
   */
  id: PropTypes.string.isRequired,

  /**
   * The row's position in the table. This is zero based.
   */
  rowIndex: PropTypes.number,

  /**
   * Data to be displayed in the cells of the row. Cells will be rendered in the row in the order given.
   */
  cells: PropTypes.arrayOf(cellShape),

  /**
   * Data for columns.
   */
  columns: PropTypes.arrayOf(formattedColumnShape).isRequired,

  /**
   * A number for column minimum width.
   */
  columnMinimumWidth: PropTypes.number,

  /**
   * A number for column maximum width.
   */
  columnMaximumWidth: PropTypes.number,

  /**
   * Number of visual columns.
   */
  numberOfColumns: PropTypes.number.isRequired,

  /**
   * Indicates if the column is flex growing column (can grow wider or shrink).
   */
  isResponsive: PropTypes.bool.isRequired,

  /**
   * Row's width in units set by widthUnit prop, such as `px`, `em`, or `rem`.
   * Disregarded if flexGrow is set to true.
   */
  rowWidth: PropTypes.number,

  /**
   * Row's maximum width for flex growing rows in units set by widthUnit prop, such as `px`, `em`, or `rem`.
   * Will have no effect if row width is set and flexGrow is omitted or false.
   */
  rowMaximumWidth: PropTypes.number,

  /**
   * Row's minimum width for flex growing rows in units set by widthUnit prop, such as `px`, `em`, or `rem`.
   * Will have no effect if row width is set and flexGrow is omitted or false.
   */
  rowMinimumWidth: PropTypes.number,

  /**
   * Callback function that will be called on click on the cell.
   */
  onCellSelect: PropTypes.func,

  /**
   * Callback function that will pass the focused column and focused row ids to the main component.
   */
  setFocusedCell: PropTypes.func,

  /**
   * The type of width value. One of `px`, `em`, `rem`. Defaults to 'px'.
   */
  widthUnit: PropTypes.oneOf([
    widthUnitTypes.PX,
    widthUnitTypes.EM,
    widthUnitTypes.REM,
  ]).isRequired,

  /**
   * A valid css string, px, em, or rem supported.
   */
  rowHeight: unitShape,

  /**
   * Indicates if the row is located in the top visual row and needs a top border.
   */
  isTopmost: PropTypes.bool,

  /**
   * Indicates if the row is located in the left visual column and needs a left border.
   */
  isLeftmost: PropTypes.bool,

  /**
   * A zero-based index indicating which column represents the row header.
   */
  rowHeaderIndex: PropTypes.number,
};

const Row = (props) => {
  const {
    id,
    rowIndex,
    cells,
    columns,
    columnMinimumWidth,
    columnMaximumWidth,
    numberOfColumns,
    isResponsive,
    rowMaximumWidth,
    rowMinimumWidth,
    rowWidth,
    widthUnit,
    onCellSelect,
    setFocusedCell,
    rowHeight,
    isTopmost,
    isLeftmost,
    rowHeaderIndex,
  } = props;

  const theme = useContext(ThemeContext);

  const style = {};
  if (isResponsive) {
    style.flex = `1 1 ${Math.min(100 / numberOfColumns)}%`;
    if (rowMaximumWidth) {
      style.maxWidth = `${rowMaximumWidth}${widthUnit}`;
    }
    if (rowMinimumWidth) {
      style.minWidth = `${rowMinimumWidth}${widthUnit}`;
    }
  } else { style.width = `${rowWidth}${widthUnit}`; }
  if (rowHeight) {
    style.height = `${rowHeight?.value}${rowHeight?.unitType}`;
  }

  const activeRow = cells && cells.length > 0;
  const rowIndexText = `Row ${rowIndex + 1}`;

  return (
    <div
      id={id}
      data-row-id={id}
      aria-rowindex={rowIndex + 2}
      // eslint-disable-next-line jsx-a11y/aria-props
      aria-rowindextext={rowIndexText}
      role={activeRow && 'row'}
      aria-hidden={activeRow ? null : true}
      className={cx('row', isTopmost && 'row-topmost', isLeftmost && 'row-leftmost', !activeRow && 'row-placeholder', theme.className)}
      // eslint-disable-next-line react/forbid-dom-props
      style={style}
    >
      {activeRow && cells.map((cellData, index) => (
        <Cell
          key={`row-${id}-col-${columns[index].id}`}
          rowIndexText={rowIndexText}
          isSelected={cellData.isSelected}
          rowId={id}
          isRowHeader={index === rowHeaderIndex}
          column={columns[index]}
          columnMinimumWidth={columnMinimumWidth}
          columnMaximumWidth={columnMaximumWidth}
          widthUnit={widthUnit}
          onCellSelect={onCellSelect}
          setFocusedCell={setFocusedCell}
        >
          {cellData.content}
        </Cell>
      ))}
    </div>
  );
};

Row.propTypes = propTypes;

export default React.memo(Row);
