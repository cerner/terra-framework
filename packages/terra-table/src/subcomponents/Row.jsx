import React, { useContext, useState } from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import ThemeContext from 'terra-theme-context';

import Cell from './Cell';
import cellShape from '../proptypes/cellShape';
import columnShape from '../proptypes/columnShape';
import RowSelectionCell from './RowSelectionCell';
import styles from './Row.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * An identifier to uniquely identify the row within the table.
   */
  id: PropTypes.string.isRequired,

  /**
   * Unique identifier for the parent table
   */
  tableId: PropTypes.string.isRequired,

  /**
   * The row's position in the table. This is zero based.
   */
  rowIndex: PropTypes.number,

  /**
   * An identifier for the section.
   */
  sectionId: PropTypes.string,

  /**
   * String that specifies height of the row. Any valid CSS width value is accepted.
  */
  height: PropTypes.string,

  /**
   * String that specifies the minimum height for the rows on the table. rowHeight takes precedence if valid CSS value is passed.
   * With this property the height of the cell will grow to fit the cell content.
   */
  rowMinimumHeight: PropTypes.string,

  /**
   * Data to be displayed in the cells of the row. Cells will be rendered in the row in the order given.
   */
  cells: PropTypes.arrayOf(cellShape),

  /**
   * A boolean indicating whether or not the row should render as selected.
   */
  isSelected: PropTypes.bool,

  /**
   * Boolean specifying whether zebra striping is enabled.
   */
  isTableStriped: PropTypes.bool,

  /**
   * A string identifier used to describe the row contents. This value will be used to construct additional labels
   * for internal controls (e.g. row selection cells).
   */
  ariaLabel: PropTypes.string,

  /**
   * Enables row selection capabilities for the table.
   * Use 'single' for single row selection and 'multiple' for multi-row selection.
   */
  rowSelectionMode: PropTypes.string,

  /**
   * All columns currently displayed.
   */
  displayedColumns: PropTypes.arrayOf(columnShape),

  /**
   * Callback function that will be called when a cell in the row is selected.
   * @param {string} rowId rowId
   * @param {string} columnId columnId
   * @param {object} event event
   */
  onCellSelect: PropTypes.func,

  /**
   * A zero-based index indicating which column represents the row header.
   * Index can be set to -1 if row headers are not required.
   */
  rowHeaderIndex: PropTypes.number,

  /**
   * @private
   * Id of the first row in table
   */
  firstRowId: PropTypes.string,

  /**
   * @private
   * Id of the last row in table
   */
  lastRowId: PropTypes.string,
};

const defaultProps = {
  rowHeaderIndex: 0,
  isSelected: false,
};

function Row(props) {
  const {
    rowIndex,
    height,
    rowSelectionMode,
    id,
    tableId,
    sectionId,
    isSelected,
    isTableStriped,
    cells,
    ariaLabel,
    displayedColumns,
    rowHeaderIndex,
    onCellSelect,
    rowMinimumHeight,
    firstRowId,
    lastRowId,
  } = props;

  const theme = useContext(ThemeContext);

  const [isHovered, setHovered] = useState(false);

  const isMultiRowSelect = (rowSelectionMode === 'multiple');
  const columnIndexOffSet = isMultiRowSelect ? 1 : 0;

  // Added to check if rowHeight is defined, it will take precedence. Otherwise the minimum row height would be used.
  const heightProperties = (height) ? {
    height,
  } : { minHeight: rowMinimumHeight };

  return (
    <tr
      aria-rowindex={rowIndex}
      data-row-id={id}
      className={cx('row', {
        selected: isSelected,
        selectable: !!rowSelectionMode,
        'striped-table-row': isTableStriped,
      }, theme.className)}
      // eslint-disable-next-line react/forbid-dom-props
      style={{ ...heightProperties }}
      onMouseEnter={rowSelectionMode ? () => { setHovered(true); } : undefined}
      onMouseLeave={rowSelectionMode ? () => { setHovered(false); } : undefined}
    >
      {
        isMultiRowSelect
        && (
        <RowSelectionCell
          rowId={id}
          columnId={displayedColumns[0].id}
          rowIndex={rowIndex}
          columnIndex={0}
          sectionId={sectionId}
          tableId={tableId}
          isSelected={isSelected}
          isHighlighted={isHovered || isSelected}
          ariaLabel={ariaLabel}
          onCellSelect={onCellSelect}
        />
        )
      }
      {cells.map((cellData, index) => {
        const cellColumnIndex = index + columnIndexOffSet;
        const columnId = displayedColumns[cellColumnIndex].id;
        return (
          <Cell
            rowId={id}
            columnId={columnId}
            rowIndex={rowIndex}
            columnIndex={cellColumnIndex}
            sectionId={sectionId}
            tableId={tableId}
            key={`${id}_${columnId}`}
            isSelected={!rowSelectionMode && cellData.isSelected}
            isMasked={cellData.isMasked}
            maskedLabel={cellData.maskedLabel}
            isSelectable={cellData.isSelectable}
            isRowHeader={rowHeaderIndex !== -1 && cellColumnIndex === (rowHeaderIndex + columnIndexOffSet)}
            isHighlighted={isHovered || isSelected}
            onCellSelect={onCellSelect}
            height={height}
            rowMinimumHeight={rowMinimumHeight}
            rowHeaderIndex={rowHeaderIndex}
            firstRowId={firstRowId}
            lastRowId={lastRowId}
            columnHighlightColor={displayedColumns[cellColumnIndex].columnHighlightColor}
            rowSelectionMode={rowSelectionMode}
          >
            {cellData.content}
          </Cell>
        );
      })}
    </tr>
  );
}

Row.propTypes = propTypes;
Row.defaultProps = defaultProps;

export default React.memo(Row);
