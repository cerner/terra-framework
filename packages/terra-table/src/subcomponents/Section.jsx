/* eslint-disable jsx-a11y/control-has-associated-label */

import React, { useCallback, useContext } from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import SectionHeader from 'terra-section-header';
import ThemeContext from 'terra-theme-context';
import GridContext, { GridConstants } from '../utils/GridContext';
import { SUBSECTION_HEADER_LEVEL } from '../utils/constants';

import Row from './Row';
import rowShape from '../proptypes/rowShape';
import subsectionShape from '../proptypes/subsectionShape';
import columnShape from '../proptypes/columnShape';
import styles from './Section.module.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * An identifier for the section. This identifier should be unique across the set of sections provided to the DataGrid.
   */
  id: PropTypes.string.isRequired,

  /**
   * Unique identifier for the parent table
   */
  tableId: PropTypes.string.isRequired,

  /**
   * The section's row index position in the table.
   */
  sectionRowIndex: PropTypes.number,

  /**
   * A boolean indicating whether or not the the section is collapsible. If true, the DataGrid's `onRequestSectionCollapse`
   * function will be called upon selection of the section header, and an icon indicating collapsibility will be rendered within the seaction header.
   */
  isCollapsible: PropTypes.bool,

  /**
   * A boolean indicating whether or not the section is collapsed. If true, the DataGrid will not render the contents of the section.
   */
  isCollapsed: PropTypes.bool,

  /**
   * A Boolean indicating whether or not the section header is visible.
   */
  isHidden: PropTypes.bool,

  /**
   * A text string to render within the section header.
   */
  text: PropTypes.string,

  /**
   * An array of row objects to be rendered within the section.
   */
  rows: PropTypes.arrayOf(rowShape),

  /**
   * An array of subsection objects to be rendered within the section.
   */
  subsections: PropTypes.arrayOf(subsectionShape),

  /**
   * Enables row selection capabilities for the table.
   * Use 'single' for single row selection and 'multiple' for multi-row selection.
   */
  rowSelectionMode: PropTypes.string,

  /**
   * All columns currently displayed.
   */
  displayedColumns: PropTypes.arrayOf(columnShape),

  /**
   * Callback function that will be called when a cell in the row is selected.
   * @param {string} rowId rowId
   * @param {string} columnId columnId
   * @param {object} event event
   */
  onCellSelect: PropTypes.func,

  /**
   * String that specifies the height for the rows in the table. Any valid CSS value is accepted.
   */
  rowHeight: PropTypes.string,

  /**
   * String that specifies the minimum height for the rows on the table. rowHeight takes precedence if valid CSS value is passed.
   * With this property the height of the cell will grow to fit the cell content.
   */
  rowMinimumHeight: PropTypes.string,

  /**
   * A zero-based index indicating which column represents the row header.
   * Index can be set to -1 if row headers are not required.
   */
  rowHeaderIndex: PropTypes.number,

  /**
   * Boolean specifying whether zebra striping is enabled.
   */
  isTableStriped: PropTypes.bool,

  /**
   * Function that is called when a collapsible section is selected. Parameters: `onSectionSelect(sectionId)`
   */
  onSectionSelect: PropTypes.func,
  /**
   * Width to assign for the section header in order to enable sticky behavior.
   * This should be a dynamic size based on the current screen width and the state of the table container's client width.
   */
  boundingWidth: PropTypes.number,
  /**
   * @private
   * Id of the first row in table
  */
  firstRowId: PropTypes.string,

  /**
    * @private
    * Id of the last row in table
    */
  lastRowId: PropTypes.string,
};

const defaultProps = {
  isCollapsible: false,
  isCollapsed: false,
  isHidden: false,
  rows: [],
};

function Section(props) {
  const {
    id,
    tableId,
    sectionRowIndex,
    isCollapsible,
    isCollapsed,
    isHidden,
    isTableStriped,
    text,
    rowSelectionMode,
    displayedColumns,
    onCellSelect,
    rowHeight,
    rowHeaderIndex,
    rows,
    subsections,
    onSectionSelect,
    rowMinimumHeight,
    boundingWidth,
    firstRowId,
    lastRowId,
  } = props;

  const theme = useContext(ThemeContext);

  const gridContext = useContext(GridContext);
  const isGridContext = gridContext.role === GridConstants.GRID;

  const hasSectionButton = isCollapsible && onSectionSelect;

  const handleClick = useCallback(() => {
    onSectionSelect(id);
  }, [id, onSectionSelect]);

  const sectionHeader = (
    <tbody>
      {/* Render section rows */}
      {!isHidden && (
      <tr
        aria-rowindex={sectionRowIndex}
        className={cx('header-row', theme.className)}
        data-section-id={id}
      >
        <th
          id={`${tableId}-${id}`}
          className={cx('header-cell')}
          align="left"
          colSpan={displayedColumns.length}
          role="columnheader"
          scope="col"
          tabIndex={isGridContext && !hasSectionButton ? -1 : undefined}
        >
          <SectionHeader
            className={cx('section-header')}
            text={text}
            isOpen={hasSectionButton ? !isCollapsed : undefined}
            onClick={hasSectionButton ? handleClick : undefined}
            isTitleSticky
            boundedWidth={boundingWidth}
          />
        </th>
      </tr>
      )}
    </tbody>
  );

  const rowProps = {
    displayedColumns,
    firstRowId,
    height: rowHeight,
    isTableStriped,
    lastRowId,
    onCellSelect,
    rowHeaderIndex,
    rowMinimumHeight,
    rowSelectionMode,
    sectionId: !isHidden ? id : undefined,
    tableId,
  };

  if (subsections) {
    return (
      <>
        {sectionHeader}
        {!isCollapsed && subsections.map((subsection) => (
          <>
            <tbody className={cx('subsection', {
              collapsed: isCollapsed,
              collapsible: isCollapsible,
            }, theme.className)}
            >

              <tr
                aria-rowindex={subsection.subSectionRowIndex}
                className={cx('header-row', theme.className)}
                data-subsection-id={subsection.id}
              >
                <th
                  id={`${tableId}-${id}-${subsection.id}`}
                  className={cx('header-cell')}
                  align="left"
                  colSpan={displayedColumns.length}
                  role="columnheader"
                  scope="col"
                  tabIndex={isGridContext ? -1 : undefined}
                >

                  <SectionHeader
                    className={cx('subsection-header-row')}
                    text={subsection.text}
                    isTitleSticky
                    boundedWidth={boundingWidth}
                    level={SUBSECTION_HEADER_LEVEL}
                  />

                </th>
              </tr>
            </tbody>
            <tbody className={cx('subsection-rows', {
              collapsed: isCollapsed,
              collapsible: isCollapsible,
            }, theme.className)}
            >
              {/* Render subsection rows */}
              {!isCollapsed && subsection.rows.map((row, rowIndex) => (
                <Row
                  rowIndex={subsection.subSectionRowIndex + (rowIndex + 1)}
                  key={row.id}
                  id={row.id}
                  cells={row.cells}
                  ariaLabel={row.ariaLabel}
                  isSelected={row.isSelected}
                  subsectionId={!isHidden ? subsection.id : undefined}
                  {...rowProps}
                />
              ))}
            </tbody>
          </>
        ))}
      </>
    );
  }

  return (
    <>
      {sectionHeader}
      <tbody className={cx('section', {
        collapsed: isCollapsed,
        collapsible: isCollapsible,
      }, theme.className)}
      >
        {/* Render section rows */}
        {!isCollapsed && rows.map((row, rowIndex) => (
          <Row
            rowIndex={sectionRowIndex + (rowIndex + 1)}
            key={row.id}
            id={row.id}
            cells={row.cells}
            ariaLabel={row.ariaLabel}
            isSelected={row.isSelected}
            {...rowProps}
          />
        ))}
      </tbody>
    </>
  );
}

Section.propTypes = propTypes;
Section.defaultProps = defaultProps;

export default React.memo(Section);
