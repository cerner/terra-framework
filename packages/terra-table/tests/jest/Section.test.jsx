import React from 'react';
import Section from '../../src/subcomponents/Section';
import Row from '../../src/subcomponents/Row';

// Source data for tests
const tableData = {
  cols: [
    { id: 'Column-0', displayName: ' Vitals' },
    { id: 'Column-1', displayName: 'March 16' },
    { id: 'Column-2', displayName: 'March 17', isSelectable: false },
  ],
  rows: [
    {
      id: '1',
      cells: [
        { content: 'Heart Rate Monitored (bpm)', isSelectable: false },
        { content: '' },
        { content: '66', isMasked: true },
      ],
    },
    {
      id: '2',
      cells: [
        { content: 'Temperature Oral (degC)' },
        { content: '36.7' },
        { content: '36.9', isMasked: true },
      ],
    },
    {
      id: '3',
      cells: [
        { content: 'Cardiac Index (L/min/m2)' },
        { content: '2.25' },
        { content: '2.28', isMasked: true },
      ],
    },
    {
      id: '4',
      isSelected: true,
      cells: [
        { content: 'Oxygen Flow Rate (L/min)' },
        { content: '63' },
        { content: '47' },
      ],
    },
  ],
};

describe('Section', () => {
  it('verifies that the table created is consistent with the rows and overflowColumns props', () => {
    const wrapper = enzyme.shallow(
      <Section
        id="test-section"
        tableId="test-table"
        text="Test Section"
        sectionRowIndex={1}
        displayedColumns={tableData.cols}
        rows={tableData.rows}
      />,
    );

    // The number of rows should match the given data.
    expect(wrapper.find(Row)).toHaveLength(tableData.rows.length);
  });

  it('verifies the rows are created with the right props', () => {
    const verifyRow = (rowIndex, rowComponent, data, overflowColumns) => {
      expect(rowComponent.props.displayedColumns).toEqual(overflowColumns);
      expect(rowComponent.props.rowSelectionMode).toBeUndefined();
      expect(rowComponent.props.tableId).toBe('test-table');
      expect(rowComponent.key).toEqual(data.id);
      expect(rowComponent.props.onCellSelect).toBeUndefined();
      expect(rowComponent.props.rowHeaderIndex).toEqual(0);
      expect(rowComponent.props.rowIndex).toEqual(rowIndex + 2);
      expect(rowComponent.props.cells).toEqual(data.cells);
    };

    const wrapper = enzyme.shallow(
      <Section
        id="test-section"
        tableId="test-table"
        sectionRowIndex={1}
        text="Test Section"
        displayedColumns={tableData.cols}
        rows={tableData.rows}
      />,
    );

    const rows = wrapper.find(Row);
    expect(rows).toHaveLength(tableData.rows.length);
    verifyRow(0, rows.get(0), tableData.rows[0], tableData.cols);
    verifyRow(1, rows.get(1), tableData.rows[1], tableData.cols);
    verifyRow(2, rows.get(2), tableData.rows[2], tableData.cols);
  });

  it('verifies the rows receive the correct props when table is zebra striped', () => {
    const wrapper = enzyme.shallow(
      <Section
        id="test-section"
        tableId="test-table"
        sectionRowIndex={1}
        text="Test Section"
        displayedColumns={tableData.cols}
        rows={tableData.rows}
        isTableStriped
      />,
    );

    const testRow = wrapper.find(Row).at(0);
    expect(testRow.props().isTableStriped).toBe(true);
    expect(testRow.props().tableId).toBe('test-table');
  });
});
