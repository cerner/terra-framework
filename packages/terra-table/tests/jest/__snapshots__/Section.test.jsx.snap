// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section verifies that the table created is consistent with the rows and overflowColumns props 1`] = `
<tbody
  className="section"
>
  <tr
    aria-rowindex={1}
    className="header"
    data-section-id="test-section"
  >
    <th
      align="left"
      colSpan={3}
      id="test-table-test-section"
      role="columnheader"
      scope="col"
    >
      <SectionHeader
        isOpen={false}
        isTitleFixed={true}
        isTransparent={false}
        level={2}
        text="Test Section"
      />
    </th>
  </tr>
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Heart Rate Monitored (bpm)",
          "isSelectable": false,
        },
        Object {
          "content": "",
        },
        Object {
          "content": "66",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="1"
    key="1"
    rowIndex={2}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Temperature Oral (degC)",
        },
        Object {
          "content": "36.7",
        },
        Object {
          "content": "36.9",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="2"
    key="2"
    rowIndex={3}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Cardiac Index (L/min/m2)",
        },
        Object {
          "content": "2.25",
        },
        Object {
          "content": "2.28",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="3"
    key="3"
    rowIndex={4}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Oxygen Flow Rate (L/min)",
        },
        Object {
          "content": "63",
        },
        Object {
          "content": "47",
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="4"
    isSelected={true}
    key="4"
    rowIndex={5}
    sectionId="test-section"
    tableId="test-table"
  />
</tbody>
`;

exports[`Section verifies the rows are created with the right props 1`] = `
<tbody
  className="section"
>
  <tr
    aria-rowindex={1}
    className="header"
    data-section-id="test-section"
  >
    <th
      align="left"
      colSpan={3}
      id="test-table-test-section"
      role="columnheader"
      scope="col"
    >
      <SectionHeader
        isOpen={false}
        isTitleFixed={true}
        isTransparent={false}
        level={2}
        text="Test Section"
      />
    </th>
  </tr>
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Heart Rate Monitored (bpm)",
          "isSelectable": false,
        },
        Object {
          "content": "",
        },
        Object {
          "content": "66",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="1"
    key="1"
    rowIndex={2}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Temperature Oral (degC)",
        },
        Object {
          "content": "36.7",
        },
        Object {
          "content": "36.9",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="2"
    key="2"
    rowIndex={3}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Cardiac Index (L/min/m2)",
        },
        Object {
          "content": "2.25",
        },
        Object {
          "content": "2.28",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="3"
    key="3"
    rowIndex={4}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Oxygen Flow Rate (L/min)",
        },
        Object {
          "content": "63",
        },
        Object {
          "content": "47",
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="4"
    isSelected={true}
    key="4"
    rowIndex={5}
    sectionId="test-section"
    tableId="test-table"
  />
</tbody>
`;

exports[`Section verifies the rows receive the correct props when table is zebra striped 1`] = `
<tbody
  className="section"
>
  <tr
    aria-rowindex={1}
    className="header"
    data-section-id="test-section"
  >
    <th
      align="left"
      colSpan={3}
      id="test-table-test-section"
      role="columnheader"
      scope="col"
    >
      <SectionHeader
        isOpen={false}
        isTitleFixed={true}
        isTransparent={false}
        level={2}
        text="Test Section"
      />
    </th>
  </tr>
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Heart Rate Monitored (bpm)",
          "isSelectable": false,
        },
        Object {
          "content": "",
        },
        Object {
          "content": "66",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="1"
    isTableStriped={true}
    key="1"
    rowIndex={2}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Temperature Oral (degC)",
        },
        Object {
          "content": "36.7",
        },
        Object {
          "content": "36.9",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="2"
    isTableStriped={true}
    key="2"
    rowIndex={3}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Cardiac Index (L/min/m2)",
        },
        Object {
          "content": "2.25",
        },
        Object {
          "content": "2.28",
          "isMasked": true,
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="3"
    isTableStriped={true}
    key="3"
    rowIndex={4}
    sectionId="test-section"
    tableId="test-table"
  />
  <Memo(Row)
    cells={
      Array [
        Object {
          "content": "Oxygen Flow Rate (L/min)",
        },
        Object {
          "content": "63",
        },
        Object {
          "content": "47",
        },
      ]
    }
    displayedColumns={
      Array [
        Object {
          "displayName": " Vitals",
          "id": "Column-0",
        },
        Object {
          "displayName": "March 16",
          "id": "Column-1",
        },
        Object {
          "displayName": "March 17",
          "id": "Column-2",
          "isSelectable": false,
        },
      ]
    }
    id="4"
    isSelected={true}
    isTableStriped={true}
    key="4"
    rowIndex={5}
    sectionId="test-section"
    tableId="test-table"
  />
</tbody>
`;
